packages/core/src/textures/BaseTexture.ts

```diff
@@ -41,7 +59,33 @@ const defaultBufferOptions = {
  */
 export class BaseTexture extends EventEmitter
 {
-    constructor(resource = null, options = null)
+    readonly width: number;
+    readonly height: number;
+    readonly resolution: number;
+    alphaMode?: ALPHA_MODES;
+    mipmap?: MIPMAP_MODES;
+    anisotropicLevel?: number;
+    scaleMode?: SCALE_MODES;
+    wrapMode?: WRAP_MODES;
+    format?: FORMATS;
+    type?: TYPES;
+    target?: TARGETS;
+    protected uid: number;
+    touched: number;
+    isPowerOfTwo: boolean;
+
+    _glTextures: { [key: number]: GLTexture };
+    dirtyId: number;
+    dirtyStyleId: number;
+    cacheId: string;
+    readonly valid: boolean;
+    readonly textureCacheIds: Array<string>;
+    readonly destroyed: boolean;
+    resource: Resource;
+    _batchEnabled: number;
+    _batchLocation: number;
+
+    constructor(resource: Resource | ImageSource | string | any = null, options: IBaseTextureOptions = null)
```

Comment:
I dont think the `any` is needed here

Reply:
its needed because its gets passed to autoDetectResource, the method can accept any possible trash if user adds more resources.

Reply:
I'm guessing this would be affected by whatever you and @bigtimebuddy decided about `autoDetectResource` below?

Reply:
yep